### PR TITLE
Feat/search profiles by name

### PR DESCRIPTION
**Pull request by hansac91**

### CODE REVIEW

File: src/app/(profile)/(components)/ProfileCard/ProfileCard.module.scss [hunk 1]
```diff
@@ -74,7 +74,6 @@
 }
 
 .technology span {
-  // min-width: 75px;
   background: #161a1f;
   text-align: center;
```
Comment (wawrzonnn) [inline src/app/(profile)/(components)/ProfileCard/ProfileCard.module.scss:78]: comment to remove

File: src/app/(profile)/(components)/ProfileCard/ProfileCard.tsx [hunk 2]
```diff
@@ -16,11 +16,44 @@ interface ProfileCardProps {
...
+const ProfileCard = ({
+  data,
+  onClick,
+  withStateStatus,
+  searchTerm,
+}: ProfileCardProps) => {
+  console.log('Search term in ProfileCard:', searchTerm)
+  const highlightText = (text: string, searchText?: string) => {
+    console.log('Highlighting:', text, 'with:', searchText)
+    if (!searchText || !searchText.trim()) {
```
Comment (wawrzonnn) [inline src/app/(profile)/(components)/ProfileCard/ProfileCard.tsx:33]: logs to remove

File: src/assets/icons/CloseIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const CloseIcon = () => {
+  return (
+    <svg
```
Comment (wawrzonnn) [inline src/assets/icons/CloseIcon.tsx:3]: it should be clear icon

**Review by wawrzonnn — APPROVED** [review]
Good Job

I think we should provide some feedback to the user when no profiles are found. Could you please check what message we currently display in the technology dropdown?

File: src/app/(profile)/(components)/ProfileCard/ProfileCard.tsx [hunk 4]
```diff
@@ -16,11 +16,44 @@ interface ProfileCardProps {
...
-const ProfileCard = ({ data, onClick, withStateStatus }: ProfileCardProps) => {
+const ProfileCard = ({
+  data,
+  onClick,
+  withStateStatus,
+  searchTerm,
+}: ProfileCardProps) => {
+  console.log('Search term in ProfileCard:', searchTerm)
+  const highlightText = (text: string, searchText?: string) => {
+    console.log('Highlighting:', text, 'with:', searchText)
```
Comment (Sumick) [inline src/app/(profile)/(components)/ProfileCard/ProfileCard.tsx:32]: **nit:** console log to remove

File: src/app/(profile)/(components)/ProfileList/ProfilesWithFilter.tsx [hunk 5]
```diff
@@ -23,19 +23,37 @@ export const ProfilesWithFilter: React.FC<{
...
+    searchTermFilter,
   } = useFilters()
   const filteredProfileData = (data || [])
     .filter(filterByPosition(jobSpecializationFilter))
     .filter(filterBySeniority(seniorityFilter))
     .filter(filterByLocation(locationFilter))
     .filter(filterByTechnology(technologyFilter))
     .filter(filterByAvailability(availabilityFilter))
+    .filter((profile) =>
+      profile.fullName.toLowerCase().includes(searchTermFilter.toLowerCase()),
```
Comment (Sumick) [inline src/app/(profile)/(components)/ProfileList/ProfilesWithFilter.tsx:35]: **nit:** we should move it to separate function, same as above, so: `filterByFullName`

Comment (Sumick) [reply src/app/(profile)/(components)/ProfileList/ProfilesWithFilter.tsx:35]: oh, there is such function already - let's just use it

**Review by Sumick — APPROVED** [review]
Good job! Code looks good - let's do little cleanup before merging

**Pull request merged**